### PR TITLE
log and skip backups for non-existent & hostPath volumes

### PR DESCRIPTION
**Pull request by skriss**
Signed-off-by: Steve Kriss <steve@heptio.com>

### CODE REVIEW

File: pkg/restic/backupper.go [hunk 1]
```diff
@@ -170,6 +182,26 @@ ForEachVolume:
 	return volumeSnapshots, errs
 }
 
+func volumeExists(pod *corev1api.Pod, volumeName string) bool {
```
Comment (ncdc) [inline pkg/restic/backupper.go]: Is it worth optimizing this into a single function so we don't have to range over pod.Spec.Volumes twice?

Comment (ncdc) [reply pkg/restic/backupper.go]:
```go
func resticCanBackup(log logrus.FieldLogger, pod *corev1api.Pod, volumeName string) bool {
  for _, volume := range pod.Spec.Volumes {
    if volume.Name == volumeName {
      if volume.HostPath != nil {
        // log that it's hostPath
        return false
      }
      return true
    }
  }
  // log that we couldn't find it
  return false
}

Comment (ncdc) [reply pkg/restic/backupper.go]: If it's easier to unit test them separately, it's fine to leave as-is. We know how to optimize and can do it later if needed.

**Comment by skriss** [comment]
switched to the map-based approach so we loop through all pods in the volume only once total, rather than once or twice per volume to backup.

**Comment by ncdc** [comment]
LGTM

**Pull request merged**